cli/include/ignition/utils/cli/IgnitionFormatter.hpp

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef IGNITION_UTILS_CLI_IGNITION_FORMATTER_HPP_
+#define IGNITION_UTILS_CLI_IGNITION_FORMATTER_HPP_
+
+#include <algorithm>
+#include <string>
+#include <sstream>
+#include <vector>
+#include <unordered_map>
+
+#include "ignition/utils/cli/App.hpp"
+#include "ignition/utils/cli/FormatterFwd.hpp"
+
+//////////////////////////////////////////////////
+/// \brief CLI Formatter class that implements custom Ignition-specific
+/// formatting.
+///
+/// More information on custom formatters:
+/// https://cliutils.github.io/CLI11/book/chapters/formatting.html
+class IgnitionFormatter: public CLI::Formatter {
+
+//////////////////////////////////////////////////
+public: explicit IgnitionFormatter(const CLI::App *_app)
+{
+  // find needs/needed_by for root options
+  for (const CLI::Option *opt: _app->get_options())
+  {
+    if (!opt->get_needs().empty())
+    {
+      for(const CLI::Option *op : opt->get_needs())
```

Comment:
`op` and `opt` are too similar. Consider renaming one of them to make this easier to read.